basket/news/utils.py

```diff
@@ -95,14 +100,66 @@ def email_is_blocked(email):
     return False
 
 
+def is_authorized(request, email=None):
+    if has_valid_api_key(request):
+        return True
+
+    if email and has_valid_fxa_oauth(request, email):
+        return True
+
+    return False
+
+
 def has_valid_api_key(request):
     # The API key could be the query parameter 'api-key' or the
     # request header 'X-api-key'.
-
     api_key = (request.REQUEST.get('api-key', None) or
                request.REQUEST.get('api_key', None) or
                request.META.get('HTTP_X_API_KEY', None))
-    return APIUser.is_valid(api_key)
+    if api_key:
+        return APIUser.is_valid(api_key)
+
+    return False
+
+
+def has_valid_fxa_oauth(request, email):
+    if not email:
+        return False
+
+    server_urls = fxa.constants.ENVIRONMENT_URLS.get(settings.FXA_OAUTH_SERVER_ENV)
+    if not server_urls:
+        return False
+
+    # Grab the token out of the Authorization header
+    authorization = request.META.get('HTTP_AUTHORIZATION')
+    if not authorization:
+        return False
+
+    authorization = authorization.split(None, 1)
+    if authorization[0].lower() != 'bearer' or len(authorization) != 2:
+        return False
```

Comment:
Do you think this header parsing is something we could provide via a more handy helper function in PyFxA itself?

Reply:
Wouldn't hurt. It's not completely trivial to correctly handle all the possible values this header might have. I'd certainly have used such a thing :) It'd be nice if you could just pass in a request object to something like that, but there is no one standard python request object and I'm sure you don't want to start having to support any set of web frameworks. But a function to parse the header value and return the token or `None` would be nice.